server/views/pages/barcode/generate-barcode-image.njk

```diff
@@ -33,40 +33,43 @@
       </div>
     </div>
 
-    <div class="govuk-grid-row" id="generate-barcode-image-container">
-      <div class="govuk-grid-column-full">
-
+    {% for barcodeImage in barcodeImages %}
+      <div class="govuk-grid-row govuk-!-margin-bottom-6">
         <div class="govuk-grid-column-full barcode-address-container">
           <div class="govuk-grid-row">
             <div class="govuk-grid-column-two-thirds">
-              <img class="barcode-address-image" alt="The generated barcode image for {{ barcodeImageName }}" src="{{ barcodeImageUrl }}">
+              <img class="barcode-address-image" alt="The generated barcode image for {{ barcodeImage.barcodeImageName }}" src="{{ barcodeImage.barcodeImageUrl }}">
             </div>
           </div>
-          <div class="govuk-button-group">
+          <div class="govuk-button-group govuk-!-margin-top-2">
             {{ govukButton({
               text: "Download",
               classes: "govuk-button--secondary",
               element: "a",
-              href: barcodeImageUrl,
-              attributes: { 'target': '_blank', 'download': barcodeImageName, 'data-qa': 'download-image-button' }
+              href: barcodeImage.barcodeImageUrl,
+              attributes: { 'target': '_blank', 'download': barcodeImage.barcodeImageName, 'data-qa': 'download-image-button' }
             }) }}
             {{ govukButton({
               text: "Copy",
               classes: "govuk-button--secondary barcode-address-copy-button",
               attributes: { 'data-qa': 'copy-image-button' }
             }) }}
           </div>
-        </div>
 
+          {% if loop.length > 1 %}
```

Comment:
Who knew nunjucks has a special `loop` variable when inside a for loop? It has a number of properties on it (loop.first, loop.last, loop.index, loop.length etc), but in this case we can use it to only show the `<hr>` element when there is more than 1 barcode to display 👍